viziphant/unitary_event_analysis.py

```diff
@@ -0,0 +1,391 @@
+            different pattern hash --> 0-axis
+            different window --> 1-axis
+        Keys:
+        -----
+        Js : list of float
+            JointSurprise of different given pattern within each window.
+        indices : list of list of int
+            A list of indices of pattern within each window.
+        n_emp : list of int
+        The empirical number of each observed pattern.
+        n_exp : list of float
+            The expected number of each pattern.
+        rate_avg : list of float
+            The average firing rate of each neuron.
+    significance_level : float
+        The significance threshold used to determine which coincident events
+        are classified ad unitary events within a window.
+    bin_size : pq.Quantity
+        The size of bins for discretizing spike trains. This value should be
+        identical to the one used to generate `joint_surprise_dict`.
+    window_size : pq.Quantity
+        The size of the analysis-window. This value should be identical to the
+        one used to generate `joint_surprise_dict`.
+    window_step : pq.Quantity
+        The size of the window step. This value should be identical to th one
+        used to generate `joint_surprise_dict`.
+    **plot_params_user
+        Plotting parameters used to update the default plotting
+        parameter values.
+        Keys:
+        -----
+        events : dictionary (default: {})
+            Epochs to be marked on the time axis
+            key: epochs name as string
+            value: list of pq.Quantity
+        figsize : tuple of int (default: (12, 10))
+            The dimensions for the figure size.
+        hspace : float (default: 1)
+            The amount of height reserved for white space between subplots.
+        wspace : float (default: 0.5)
+            The amount of width reserved for white space between subplots.
+        top : float (default: 0.95)
+        bottom : float (default: 0.1)
+        right : float (default: 0.87)
+        left : float (default: 0.08)
+            The sizes of the respective margin of the subplot in the figure.
+        fsize : integer (default: 12)
+            The size of the font
+        unit_real_ids : list of integers (default: [1, 2])
+            The unit ids form the experimental recording.
+        lw: float (default: 0.5)
+            The default line width.
+        S_ylim : tuple of ints or floats (default: (-3, 3))
+            The y-axis limits for the joint surprise plot.
+        marker_size : integers (default: 5)
+            The marker size for the coincidence and unitary events.
+        y_tick_interval : integers (default: 15)
+            The size of the interval between y-ticks (for rasterplots only)
+        showfig : boolean (default: True)
+            Displays the figure on screen if True.
+        savefig : boolean (default: False)
+            Saves the figure to disk if True.
+        path_filname_format : string (default: figure.pdf)
```

Comment:
Typo: `path_filename_format`.
```suggestion
        path_filename_format : string (default: figure.pdf)
```